demo/src/screens/ProductScreen/ProductScreen.js

```diff
@@ -1,13 +1,100 @@
 import React from 'react';
+import {Container, Row, Col, Image, Button} from 'react-bootstrap';
+import {useParams} from 'react-router-dom';
+import {connect} from 'react-redux';
+import PropTypes from 'prop-types';
+import {Link} from 'react-router-dom';
+import {addOneToCart} from '../../store/StoreHelpers.js';
+
 
 /**
  * @return {!JSX} Page component for where a user can view
  *     details about a single product.
  */
-export function ProductScreen() {
+const ProductScreenBase = function({items, addToCart}) {
+  const {id} = useParams();
+
+  const currProduct = items[id];
+
+  const buttonText = currProduct.inCart ? 'In Cart' : 'Add to Cart';
+
+  const addOnClick = () => {
+    addToCart(id);
+  };
+
+  const buyOnClick = () => {
+    if (!currProduct.inCart) {
+      addToCart(id);
+    }
+  };
+
   return (
-    <div>
-      <h1>{`I'm the Product Page!`}</h1>
-    </div>
+    <Container>
+      <Row>
+        <Col/>
+        <Col xs={5}>
+          <div>
+            <Image src={currProduct.image} width="100%"/>
+            <Container>
+              <Row>
+                <Col>
+                    Name: {currProduct.name}
+                </Col>
+                <Col xs="8"/>
+                <Col>
+                    Price: ${currProduct.cost}
+                </Col>
+              </Row>
+            </Container>
+            <br/>
+            <p>
+              {currProduct.description}
+            </p>
+            <Container>
+              <Row>
+                <Col>
+                  <Button onClick={addOnClick} disabled = {currProduct.inCart}>
+                    {buttonText}
+                  </Button>
+                </Col>
+                <Col xs="3"/>
+                <Col>
+                  <Button variant="success" onClick={buyOnClick}
+                    as={Link} to='/cart'>Buy Now</Button>
+                </Col>
+              </Row>
+            </Container>
+          </div>
+        </Col>
+        <Col/>
+      </Row>
+    </Container>
   );
-}
+};
+
+ProductScreenBase.propTypes = {
+  items: PropTypes.object,
+  addToCart: PropTypes.func,
+};
+
+// pass state.items as items into ProductScreen props
+const mapStateToProps = (state) => ({
+  items: state.items,
+});
+
+// dispatch addOneToCart as addToCart to ProductScreen props
+const mapDispatchToProps = (dispatch) => {
+  return {
+    addToCart: (id) => {
+      dispatch(addOneToCart(id));
+    },
+  };
+};
+
+/*
+ * Decorate the ProductScreenBase function, implicity passing in
+ * information about the global site state. Also pass mapDispatchToProps
+ * function to allow modification of global cart.
+ */
+export const ProductScreen = connect(mapStateToProps,
```

Comment:
nit: It would enhance readability to do:

```
export const ProductScreen =
    connect(mapStateToProps, mapDispatchToProps)(ProductScreenBase);
```

Reply:
Done!